docs-web/src/main/java/com/sismics/docs/rest/resource/FileResource.java

```diff
@@ -417,7 +417,7 @@ public Response list(
         boolean authenticated = authenticate();
         
         // Check document visibility
-        if (documentId != null) {
+        if (documentId == null) {
```

Comment:
Here I am not quite sure what the intention was, but `documentId != null` is definitely not the behavior we want.

Reply:
But sharing is now not working anymore. I will look into this.

Reply:
`documentId != null` is the intended code. I don't see a problem here.

Reply:
Because if I get: `http://127.0.0.1:8080/api/file/list?id=e78be12f-2f8b-4a22-91e9-b2126983c14b` for example I do not get a `403`or a document not found but a `500`. This happens when I am not logged in. It returns an SQL-Error (`ERROR: syntax error at or near ")"`) meaning the SQL-Statement within the `aclDao.checkPermission` cannot handle this use-case in which the `targetIdList` is empty.

Maybe a posibility would be to check if the `targetIdList` is empty before checking the permissions?
```Java
// Check document visibility
if (documentId != null) {
    AclDao aclDao = new AclDao();
    List<String> targetIdList = getTargetIdList(shareId);
    if(targetIdList.isEmpty()) {
        throw new ForbiddenClientException();
    } else if (!aclDao.checkPermission(documentId, PermType.READ, targetIdList)) {
        throw new NotFoundException();
    }
} else if (!authenticated) {
    throw new ForbiddenClientException();
}
```

Or should the `aclDao.checkPermission` -Method be able to handle an empty targetIdList? Then we would have to adjust that method.

### Attachments:
[Error.pdf](https://github.com/sismics/docs/files/5792573/Error.pdf)

Reply:
I think checkPermission should be modified to return false if the targetIdList is empty, it would make sense.

Reply:
If the user is not logged in at all shouldn't the service return a `403` instead of a `File not found`? I get the `File not found` in case of a guest or other user but in this situation i do not know what would be better.